src/btree/bt_curnext.c

```diff
@@ -678,16 +678,6 @@ __wt_btcur_next_prefix(WT_CURSOR_BTREE *cbt, WT_ITEM *prefix, bool truncating)
     for (newpage = false;; newpage = true, restart = false) {
         page = cbt->ref == NULL ? NULL : cbt->ref->page;
 
-        /*
```

Comment:
Removing this check means we can potentially read a page full of tombstones before the optimisation kicks in in the later part of this loop?

Reply:
yep, I think that is okay, being a single page that will avoid the optimization.